python-avd/pyavd/_eos_designs/structured_config/network_services/metadata.py

```diff
@@ -68,45 +67,43 @@ def set_cv_pathfinder_metadata_internet_exit_policies(
 
     def set_cv_pathfinder_metadata_applications(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
         """Set the metadata.cv_pathfinder.applications if available."""
-        if not self.shared_utils.is_cv_pathfinder_server or self.application_traffic_recognition is None:
+        if not self.shared_utils.is_cv_pathfinder_server or (atr := self.structured_config.application_traffic_recognition) is None:
             return
-
-        applications = get(self.application_traffic_recognition, "applications", default=[])
-        user_defined_app_names = set(get_all(applications, "ipv4_applications.name") + get_all(applications, "ipv6_applications.name"))
-        categories = get(self.application_traffic_recognition, "categories", default=[])
-        for profile in get(self.application_traffic_recognition, "application_profiles", default=[]):
-            application_profile = EosCliConfigGen.Metadata.CvPathfinder.Applications.ProfilesItem(name=profile["name"])
-            protocols = get(profile, "application_transports")
-            if protocols:
-                application_profile.transport_protocols.extend(protocols)
-            for application in get(profile, "applications", default=[]):
-                if application["name"] not in user_defined_app_names:
-                    services = get_all(application, "service")
+        applications = atr.applications
+        # applications.ipv6_applications is not present in the model. Hence corrected the line.
+        user_defined_app_names = set(applications.ipv4_applications.keys())
+        categories = atr.categories
+        for profile in atr.application_profiles:
+            application_profile = EosCliConfigGen.Metadata.CvPathfinder.Applications.ProfilesItem(name=profile.name)
+            application_profile.transport_protocols.extend(profile.application_transports)
+            for application in profile.applications:
+                if application.name not in user_defined_app_names:
+                    services = application.service or []
```

Comment:
```suggestion
                if application.name not in user_defined_app_names:
                    if application.service is not None:
```

and fix the reste below

Reply:
Fixed.